### PR TITLE
Add event system explanation to jib-core readme

### LINKED ISSUE #1462: Add detailed documentation for the jib-core events

### PR DESCRIPTION
**Pull request by TadCordle**
Fixes #1462. Should be merged after the next core release (0.1.2? 0.2.0?)

### CODE REVIEW

File: jib-core/README.md [hunk 1]
```diff
@@ -98,17 +98,42 @@ The Jib Core system consists 3 main parts:
...
+// Register with Jib
+Jib.from(...)
+    ...
+    .containerize(
+        Containerizer.to(...)
+            ...
+            .setEventHandlers(eventHandlers));
+```
+
+When Jib dispatches events, the event handlers you defined for that event type will be called. The following are the types of events you can listen for in Jib core (see [API reference](http://static.javadoc.io/com.google.cloud.tools/jib-core/0.1.1/com/google/cloud/tools/jib/api/package-summary.html) for more information):
```
Comment (loosebazooka) [inline jib-core/README.md]: can this reference a *latest* instead of 0.1.1? Seems like we could possibly miss updating this when doing releases?

Comment (loosebazooka) [reply jib-core/README.md]: Looks like the same thing through the readme.

Comment (loosebazooka) [reply jib-core/README.md]: There appears to be some provision for it at: https://www.javadoc.io/, but I don't know how we're generating these in the first place.

Comment (loosebazooka) [reply jib-core/README.md]: okay, so the url just has to be formatted a little weird

```diff
- http://static.javadoc.io/com.google.cloud.tools/jib-core/0.1.1/com/google/cloud/tools/jib/api/package-summary.html
+ https://www.javadoc.io/page/com.google.cloud.tools/jib-core/latest/com/google/cloud/tools/jib/api/package-summary.html
```

Comment (TadCordle) [reply jib-core/README.md]: Sounds like a good idea to me

**Comment by TadCordle** [comment]
Since we're releasing 0.1.2, we can review this @GoogleContainerTools/java-tools-build

**Pull request merged**